content/en/blog/trusted-boot.md

```diff
@@ -0,0 +1,155 @@
+
+This chain is the one we in the Linux and tech industry we are very much familiar with. Several designs and strategies have been developed to secure this stack over time, for instance Verified kernels and drivers, or Secure Boot to sign the bootloader and be able to trust its authenticity. There are various bootloader implementations as well, tied to more secure HW (like TPM devices).
+
+However, as technologies and security measures evolved, attacks evolved as well. Specific hardware now can be leveraged to increase the security posture of the boot process, and it is time for a change, as the market demands more sophisticated security measures to protect against malicious actors.
+
+## Booting with a TPM-equipped Hardware
+
+When a device equipped with TPM, and a sufficiently modern hardware boots, several steps happen sequentially.
+
+Many hardware platforms use a Core Root of Trust for Measurements (CRTM), which is the very first thing that boots, even before the firmware. The CRTM gets a hash of the firmware and sends it to the TPM chip, which measures the running software and is a requirement for trusted boot. The TPM chip then loads the firmware/BIOS. 
+
+The TPM takes note of the measurement and stores the hash in a bank of multiple platform configuration registers (PCRs). In order to store measurements, the TPM chip extends the banks from the previous values, as these changes to the stack are easy to recognize during the boot.
+
+![CRTM trusted boot](https://github.com/kairos-io/kairos-docs/assets/2420543/5a0244c3-ff58-4ea0-83b3-b391ccac0b91)
+
+Next, the BIOS or firmware measures the subsequent stage (bootloader or UKI) and sends it to the TPM, then loads the UKI and continues booting.
+
+Once the TPM is asked to release an encryption key to unseal the full disk encryption, it will check if the measurements it has are valid. If a bootloader is present, it will have already measured the UKI files, and measurement also happens when initrd starts and when the running system is ready to check eventual manipulations (e.g., kernel boot command line). The process is also bound to the secure boot signatures, so any manipulation of the UKI files would be allowed only by the key holders.
+
+If a malicious attacker tries to load custom software or modify an image in any way, they must extend the PCR value and align it with the value it would have had following an expected boot. The cryptographic robustness of the hash algorithm makes achieving this computationally expensive, thereby helping to support security. 
+
+If the firmware or bootloader have been tampered with, the hash values stored in the PCR won’t match expected measurements, and the system will flag an alert and prevent boot from happening — intentionally bricking the host device as a failsafe. 
+
+## Kairos and Trusted Boot
+
+![standardlinux](https://github.com/kairos-io/kairos-docs/assets/2420543/df635d48-e9ff-4759-98d0-3cdaa87ee8b5)
```

Comment:
The boot partition in this image says "unencrypted" . Is it possible to update the image to say "unencrypted but measured" instead? Just to make sure it doesn't show up as a "hole" in security.

Reply:
that image is just for standard linux systems, so it is not measured at all

Reply:
this one (which is the one refered to UKI/EFI) shows measured/verified: https://github-production-user-asset-6210df.s3.amazonaws.com/2420543/315326276-19b61dd1-b33f-4cfe-9d0a-b60198b0cfd5.png?X-Amz-Algorithm=AWS4-HMAC-SHA256&X-Amz-Credential=AKIAVCODYLSA53PQK4ZA%2F20240417%2Fus-east-1%2Fs3%2Faws4_request&X-Amz-Date=20240417T082545Z&X-Amz-Expires=300&X-Amz-Signature=ecab4a7fd4f6ec1a8b2c02134d513e1c164ac910753482dfda029ae5020b066a&X-Amz-SignedHeaders=host&actor_id=2420543&key_id=0&repo_id=632358571